libs/tangle-shared-ui/src/components/NetworkSelectorDropdown/index.tsx

```diff
@@ -17,16 +17,23 @@ import {
   Typography,
 } from '@tangle-network/ui-components';
 import { type FC, useCallback, useMemo } from 'react';
-import { twMerge } from 'tailwind-merge';
+import cx from 'classnames';
 import useNetworkStore from '../../context/useNetworkStore';
 import useSwitchNetwork from '../../hooks/useSwitchNetwork';
 import createCustomNetwork from '../../utils/createCustomNetwork';
 import { NetworkSelectorDropdown } from './NetworkSelectorDropdown';
 
-const NetworkSelectionButton: FC<{
-  isBridgePage?: boolean;
-  bridgeSourceChain?: ChainConfig;
-}> = ({ isBridgePage = false, bridgeSourceChain }) => {
+type NetworkSelectionButtonProps = {
+  disableChainSelection?: boolean;
+  preferredChain?: ChainConfig;
+  overrideIsWrongEvmNetworkWithPreferredChain?: boolean;
+};
+
+const NetworkSelectionButton: FC<NetworkSelectionButtonProps> = ({
+  disableChainSelection = false,
+  preferredChain,
+  overrideIsWrongEvmNetworkWithPreferredChain = false,
```

Comment:
I believe `preferredChain` alone is sufficient to condition the component's behavior; this prop is unnecessary, in my opinion.